src/Microsoft.Health.Client/ManagedIdentityCredentialProvider.cs

```diff
@@ -5,40 +5,43 @@
 
 using System.Threading;
 using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
 using EnsureThat;
-using Microsoft.Azure.Services.AppAuthentication;
 using Microsoft.Extensions.Options;
-using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Microsoft.Health.Client.Configuration;
 
 namespace Microsoft.Health.Client;
 
 public class ManagedIdentityCredentialProvider : CredentialProvider
 {
     private readonly IOptionsMonitor<ManagedIdentityCredentialConfiguration> _managedIdentityCredentialConfigurationMonitor;
-    private readonly IHttpClientFactory _httpClientFactory;
     private readonly string _optionsName;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="ManagedIdentityCredentialProvider"/> class.
     /// This class is used to obtain a token for the configured resource via Managed Identity.
     /// </summary>
     /// <param name="managedIdentityCredentialConfigurationMonitorMonitor">The configuration of the token to obtain.</param>
-    /// <param name="httpClientFactory">An optional <see cref="IHttpClientFactory"/> for use within the <see cref="AzureServiceTokenProvider"/>.</param>
     /// <param name="optionsName">Optional name to use when retrieving options from the IOptionsMonitor</param>
-    public ManagedIdentityCredentialProvider(IOptionsMonitor<ManagedIdentityCredentialConfiguration> managedIdentityCredentialConfigurationMonitorMonitor, IHttpClientFactory httpClientFactory = null, string optionsName = null)
+    public ManagedIdentityCredentialProvider(IOptionsMonitor<ManagedIdentityCredentialConfiguration> managedIdentityCredentialConfigurationMonitorMonitor, string optionsName = null)
     {
         EnsureArg.IsNotNull(managedIdentityCredentialConfigurationMonitorMonitor, nameof(managedIdentityCredentialConfigurationMonitorMonitor));
 
         _managedIdentityCredentialConfigurationMonitor = managedIdentityCredentialConfigurationMonitorMonitor;
         _optionsName = optionsName;
-        _httpClientFactory = httpClientFactory;
     }
 
     protected override async Task<string> BearerTokenFunction(CancellationToken cancellationToken)
     {
         ManagedIdentityCredentialConfiguration managedIdentityCredentialConfiguration = _managedIdentityCredentialConfigurationMonitor.Get(_optionsName);
 
-        var azureServiceTokenProvider = new AzureServiceTokenProvider(httpClientFactory: _httpClientFactory);
-        return await azureServiceTokenProvider.GetAccessTokenAsync(managedIdentityCredentialConfiguration.Resource, managedIdentityCredentialConfiguration.TenantId, cancellationToken);
```

Comment:
Coincidentally, I was just talking to Jared about this class yesterday. Should we be using `Azure.Identity` elsewhere? Like using `ClientCertificateCredential` for `OAuth2ClientCertificateCredential`? Or are we ensuring this is agnostic of Azure?

Reply:
We're ensuring that this is agnostic of azure to support our OSS use cases.